### PR TITLE
Show how to use additional parameters

### LINKED ISSUE #148: Provide easy access to additional diffraction engine parameters
Some diffractometer engines and modes have additional parameters, such as a second _(hkl)_.  Make it easier to set, get, and report these parameters.
Clipboard02

### PR DESCRIPTION
**Pull request by prjemian**
FIX #148

### CODE REVIEW

**Comment by prjemian** [comment]
@jwkim-anl, @gfabbris, @strempfer, @rodolakis: You might be interested in this.  See: https://github.com/bluesky/hklpy/blob/148-additional-parameters/examples/var_parameters.ipynb

**Review by mrakitin — APPROVED** [review]
I don't have much to add, looks good! I have a single comment about the naming.

File: examples/var_parameters.ipynb [hunk 1]
```diff
@@ -0,0 +1,322 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "50916cd1",
+   "metadata": {},
+   "source": [
+    "# Diffractometer \"Parameters\"\n",
+    "\n",
+    "Some of the diffractometer _modes_ use additional parameters.  The [E4CV](https://people.debian.org/~picca/hkl/hkl.html#org7ef08ba) geometry, for example, has a `double_diffraction` mode which requires a reference $hkl_2$ vector.  The vector is set and accessed by a Python command that calls directly the corresponding *libhkl* function:\n",
```
[the remark this answers is not in the document]
Comment (prjemian) [reply examples/var_parameters.ipynb:10]: You are correct.  That term is no longer obvious, let alone popular.  Checking my source for such information ([NIST publication of CODATA Fundamental Physical Constants](https://physics.nist.gov/cuu/pdf/wallet_2018.pdf)) does not show it in use at all.  I'll change it.

[the remark this answers is not in the document]
Comment (prjemian) [reply examples/var_parameters.ipynb:10]: Product of Planck's constant and speed of light in a vacuum, per [2018 CODATA table](https://physics.nist.gov/cuu/pdf/wallet_2018.pdf).  I will also update the precise number used in the code, leaving a reference in the comments.

[the remark this answers is not in the document]
Comment (prjemian) [reply examples/var_parameters.ipynb:10]: note: https://www.nist.gov/programs-projects/codata-values-fundamental-physical-constants

[the remark this answers is not in the document]
Comment (mrakitin) [reply examples/var_parameters.ipynb]: Thanks for the useful links, Pete! I'll add to your list: https://docs.scipy.org/doc/scipy/reference/constants.html.

**Comment by prjemian** [comment]
There it is (with different units): 

> quantity | value
> -- | --
> inverse meter-electron volt relationship | 1.239841984e-06 eV

**Comment by prjemian** [comment]
Noting:

> In addition to the above variables, scipy.constants also contains the 2018 CODATA recommended values

**Comment by mrakitin** [comment]
```py
$ ipython
Python 3.7.7 (default, Mar 26 2020, 15:48:22)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.21.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from scipy import constants

In [2]: constants.h
Out[2]: 6.62607015e-34

In [3]: constants.c
Out[3]: 299792458.0

In [4]: constants.e
Out[4]: 1.602176634e-19

In [5]: constants.h * constants.c / constants.e
Out[5]: 1.2398419843320026e-06

In [6]: constants.h * constants.c / constants.e / 1e-10  # in eV*Angstrom
Out[6]: 12398.419843320025

In [7]: constants.h * constants.c / constants.e / 1e-10 / 1e3 # in keV*Angstrom
Out[7]: 12.398419843320026
```

**Comment by mrakitin** [comment]
> There it is (with different units):
> 
> > quantity
> > value
> > 
> > 
> > 
> > 
> > inverse meter-electron volt relationship
> > 1.239841984e-06 eV

Yes, we can use that too:
```py
In [10]: constants.physical_constants['inverse meter-electron volt relationship']
Out[10]: (1.239841984e-06, 'eV', 0.0)

In [11]: constants.physical_constants['inverse meter-electron volt relationship'][0]
Out[11]: 1.239841984e-06

In [12]: constants.physical_constants['inverse meter-electron volt relationship'][0] * 1e10
Out[12]: 12398.41984

In [13]: constants.physical_constants['inverse meter-electron volt relationship'][0] * 1e10 / 1e3
Out[13]: 12.39841984
```
But hard-coding it is probably less error-prone.

**Comment by prjemian** [comment]
There's a second problem here. The 4 on the end causes the unit test to
fail when trying the 100,1,1 reflection. That's why I did not code that
last digit of precision.

**Comment by prjemian** [comment]
I'm happy with hard coding this number, with a comment about its origin.

**Comment by mrakitin** [comment]
@prjemian, yes, makes sense to me too. Thanks, it was a fruitful discussion.

**Pull request merged**